File: ruby/grade-school/grade_school.rb
class School

  attr_accessor :to_hash

  def initialize
    @to_hash = Hash.new(Array.new)
  end

  # def add(name, grade)
  #   to_hash.merge(Hash.new { |name, grade| name[grade] += 1 })
  # end

  def add(name, grade)
    if to_hash[grade]
      @to_hash[grade].push(name)
    else
      @to_hash[grade] = name
    end
  end

end

File: javascript/grade-school/grade-school.js
'use strict';

class School {
  constructor() {
    this.currentRoster = {};
  }

  roster() {
    return this.currentRoster;
  }

  add(name, grade) {
    if (this.currentRoster[grade]) {
      this.currentRoster[grade].push(name);
      this.currentRoster[grade].sort();
    } else {
      this.currentRoster[grade] = [name];
    }
  }

  grade(grade) {
    return this.currentRoster[grade] || [];
  }
}

module.exports = School;
